dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -236,6 +236,14 @@ protected void unsetWebMBeanRuntime(ServiceReference<WebMBeanRuntime> ref) {
     private URIMatcherFactory uriMatcherFactory;
     
     
+    public static final int SPEC_LEVEL_30 = 30;
+    public static final int SPEC_LEVEL_31 = 31;
+    public static final int SPEC_LEVEL_40 = 40;
+    public static final int SPEC_LEVEL_50 = 50;
+    private static final int DEFAULT_SPEC_LEVEL = 30;
+
+    private static int loadedContainerSpecLevel = loadServletVersion();
```

Comment:
I still think this should be final.  In the tests you reflect to change this.  That can still be done even on final static fields for testing.

Reply:
I don't think set this to final without tests failing.
I kept getting: 
```java.lang.IllegalAccessException: Can not set static final int field com.ibm.ws.webcontainer.osgi.WebContainer.loadedContainerSpecLevel to (int)30``` 

 I even tried using Field.setInt instead.

Reply:
Right, I guess it is more complicated to set static final fields.  I think you could do this:

```java
    private static void setFinalStaticField(Class<?> clazz, String fieldName, Object value)
            throws ReflectiveOperationException {

        Field field = clazz.getDeclaredField(fieldName);
        field.setAccessible(true);

        Field modifiers = Field.class.getDeclaredField("modifiers");
        modifiers.setAccessible(true);
        modifiers.setInt(field, field.getModifiers() & ~Modifier.FINAL);

        field.set(null, value);
    }
```

At any rate having the field be effectively final is probably good enough here.  Just something to consider if you do want to make it final for runtime.